MagneticField/Records/interface/IdealMagneticFieldRecord.h

```diff
@@ -6,10 +6,12 @@
 #include "CondFormats/DataRecord/interface/RunSummaryRcd.h"
 #include "CondFormats/DataRecord/interface/MagFieldConfigRcd.h"
 #include "CondFormats/DataRecord/interface/MFGeometryFileRcd.h"
+#include "Geometry/Records/interface/GeometryFileRcd.h"
 #include "boost/mpl/vector.hpp"
 
-class IdealMagneticFieldRecord : public edm::eventsetup::DependentRecordImplementation<
-                                     IdealMagneticFieldRecord,
-                                     boost::mpl::vector<MFGeometryFileRcd, RunInfoRcd, MagFieldConfigRcd> > {};
+class IdealMagneticFieldRecord
+    : public edm::eventsetup::DependentRecordImplementation<
+          IdealMagneticFieldRecord,
+          boost::mpl::vector<MFGeometryFileRcd, RunInfoRcd, GeometryFileRcd, MagFieldConfigRcd> > {};
```

Comment:
I am not proficient in record magics so this may be a silly questuion, but why do we need to add a dependency on GeometryFileRcd? I understand that MFGeometryFileRcd is the record we use to retrieve the MF geometry from the DB and I naively expect GeometryFileRcd to be the corresponding one for reco/SIM geometry; why do we need this dependency then?

Reply:
As we discussed in our meeting, the DD4hep approach requires reading the magnetic field XML files through a `DDDetector`, which requires the `GeometryFileRcd`. This new dependency doesn't create any issues.